client/src/main/java/edu/colostate/cs/cs414/chesshireCoders/jungleClient/client/GamesManager.java

```diff
@@ -33,6 +34,22 @@ public void setHomeController(HomeController homeController) {
 		this.homeController = homeController;
 	}
 
+	public void handleBoardUpdateEvent(long gameID) {
+		JungleGame game = findById(gameID);
+		
+		if (game != null)
+			getAndShowGame(gameID, game.getViewingPlayer());
+	}
+	
+	public void handleGameEndedEvent(long gameID) {
+		JungleGame game = findById(gameID);
+		
+		if (game != null)
+			getAndShowGame(gameID, game.getViewingPlayer());
+		game = findById(gameID);

```

Comment:
Why do you run findById in both like 45 and 49. Won't it just set itself to the same value?

Reply:
That first findById and game != null check would be better represented with localGameExists. The sencond findById gets the just-updated game.